src/miniscript/satisfy.rs

```diff
@@ -440,7 +455,8 @@ impl Witness {
                 ret.push(hashtype.as_u32() as u8);
                 Witness::Stack(vec![ret])
             }
-            None => Witness::Unavailable,
+            // Signatures cannot be forged
+            None => Witness::Impossible,
```

Comment:
This should stay as `Unavailable`

Reply:
after talking on IRC I think you're right actually. Our assumption is that a given satisfier knows all the signatures, and no others exist. (We assume no duplicated keys and assume our counterparties are not trying to malleate us.)